components/sup/src/manager/service/mod.rs

```diff
@@ -77,26 +81,31 @@ impl ServiceSpec {
             topology: Topology::default(),
             update_strategy: UpdateStrategy::default(),
             binds: vec![],
-            dev_config_from: None,
+            config_from: None,
         }
     }
 }
 
 #[derive(Debug, Serialize)]
 pub struct Service {
-    pub needs_restart: bool,
-    pub package: Package,
     pub cfg_incarnation: u64,
-    pub service_group: ServiceGroup,
-    pub topology: Topology,
-    pub update_strategy: UpdateStrategy,
+    pub config: ServiceConfig,
+    pub config_root: PathBuf,
     pub current_service_files: HashMap<String, u64>,
+    pub depot_url: String,
```

Comment:
I keep wanting this to be a `Url` but keep falling back to a `String`. Partly because there is no Serialize implemented on `url:Url` and partly because our codebase pre-validates this value before inserting it anyway.